sdk/eventhub/Azure.Messaging.EventHubs/tests/Primitives/EventProcessorOptionsTests.cs

```diff
@@ -0,0 +1,120 @@
+                DefaultStartingPosition = EventPosition.FromOffset(555)
+            };
+
+            EventProcessorOptions clone = options.Clone();
+            Assert.That(clone, Is.Not.Null, "The clone should not be null.");
+            Assert.That(clone, Is.Not.SameAs(options), "The options should be a copy, not the same instance.");
+
+            Assert.That(clone.ConnectionOptions.TransportType, Is.EqualTo(options.ConnectionOptions.TransportType), "The connection options of the clone should copy properties.");
+            Assert.That(clone.ConnectionOptions, Is.Not.SameAs(options.ConnectionOptions), "The connection options of the clone should be a copy, not the same instance.");
+            Assert.That(clone.RetryOptions.IsEquivalentTo(options.RetryOptions), Is.True, "The retry options of the clone should be considered equal.");
+            Assert.That(clone.RetryOptions, Is.Not.SameAs(options.RetryOptions), "The retry options of the clone should be a copy, not the same instance.");
+            Assert.That(clone.MaximumWaitTime, Is.EqualTo(options.MaximumWaitTime), "The maximum wait time should match.");
+            Assert.That(clone.PrefetchCount, Is.EqualTo(options.PrefetchCount), "The prefetch count should match.");
+            Assert.That(clone.LoadBalancingUpdateInterval, Is.EqualTo(options.LoadBalancingUpdateInterval), "The load balancing update interval should match.");
+            Assert.That(clone.Identifier, Is.EqualTo(options.Identifier), "The identifier should match.");
+            Assert.That(clone.TrackLastEnqueuedEventProperties, Is.EqualTo(options.TrackLastEnqueuedEventProperties), "Tracking of last enqueued events should match.");
+            Assert.That(clone.DefaultStartingPosition, Is.EqualTo(options.DefaultStartingPosition), "The default starting position should match.");
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="EventProcessorOptions.ConnectionOptions" />
+        ///   property.
+        /// </summary>
+        ///
+        [Test]
+        public void ConnectionOptionsAreValidated()
+        {
+            Assert.That(() => new EventProcessorOptions { ConnectionOptions = null }, Throws.ArgumentNullException);
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="EventProcessorOptions.RetryOptions" />
+        ///   property.
+        /// </summary>
+        ///
+        [Test]
+        public void RetryOptionsAreValidated()
+        {
+            Assert.That(() => new EventProcessorOptions { RetryOptions = null }, Throws.ArgumentNullException);
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="EventProcessorOptions.MaximumWaitTime" />
+        ///   property.
+        /// </summary>
+        ///
+        [Test]
+        [TestCase(-1)]
+        [TestCase(-10)]
+        [TestCase(-100)]
+        public void MaximumWaitTimeIsValidated(int waitTimeSeconds)
```

Comment:
Is `0` a valid wait time? Same question for the others.

Reply:
It is.  In a nutshell, that would indicate that the caller wanted a batch of whatever was available in the prefetch queue and did not want to wait for any additional network activity to fetch more.    This isn't an unusual case, as it is used in concert with the prefetch count option to tune throughput scenarios.